### DISCUSSION TITLE
log4net crashes while creating mutex

### DISCUSSION THREAD

**Discussion by tbald-o** [apache/logging-log4net#218] [General]
We have a .NET Core 3.1 application running on Linux where we use log4net 2.0.12.
In the last two weeks, we run into this exception twice:

```
Unhandled exception. System.IO.IOException: The system cannot open the device or file specified. : '_opt_Dists_2025.1.1.h1_WorkingDirectories_8255_b44c8ce2-574b-48a2-8c4d-2aee273f9bf5_app.log'
   at System.Threading.Mutex.CreateMutexCore(Boolean initiallyOwned, String name, Boolean& createdNew)
   at System.Threading.Mutex..ctor(Boolean initiallyOwned, String name)
   at log4net.Appender.RollingFileAppender.ActivateOptions()
```

Some additional info:

- log folder name is unique since it is created at runtime using a guid (so mutex name should be unique too)
- log file is not even created
- bug is not deterministic, if we re run our app, everything goes fine

**Comment by FreeAndNil** [comment]
Hi @strpp please test with our latest version: https://github.com/apache/logging-log4net/releases/tag/rel%2F3.0.3 and **.net 8**.
log4net 2.x is no longer supported.
.Net Core 3.1 is no longer supported for more than two years.

**Reply by tbald-o** [reply]
Thanks for the reply but unfortunately we cannot update to .NET 8. 
We'll try to use 2.0.17 version and later (if possible) we'll update to 3.0.3

**Reply by tbald-o** [reply]
Also, mutex creation in _RollingFileAppender.ActivateOptions_ seems to not have changed in further versions (except for that _rolling appended to the mutex name) so I'm afraid updating won't solve the issue

**Reply by FreeAndNil** [reply]
@strpp the Mutex ctor can fail for many reasons on Linux (see https://github.com/PowerShell/PSReadLine/issues/1464).
There are probably fixes (or better error messages) in the supported .net versions.
